frontend/viewer/src/lib/layout/Scotty.svelte

```diff
@@ -0,0 +1,43 @@
+<script lang="ts" context="module">
+  const portals = ['right-toolbar', 'app-bar-menu'] as const;
+  type ScottyPortalTarget = typeof portals[number];
+
+  export function initScottyPortalContext(): void {
+    portals.forEach((name) => {
+      setContext(name, writable<HTMLElement>());
+    });
+  }
+
+  export function asScottyPortal(elem: HTMLElement, name: ScottyPortalTarget): void {
+    const scottyPortal = useScottyPortal(name) as Writable<HTMLElement>;
+    scottyPortal.set(elem);
```

Comment:
we should setup a destroy function to remove the element from the portal when it's gone. That will mean that the store value could be undefined, but we could just log a warning when trying to render something to a portal which doesn't have an outlet

Reply:
I added a destroy as part of the action return. I'm not sure about logging a warning. The app bar menu portal is in a popover menu, so it only exists if the menu is open.